python/ray/data/dataset.py

```diff
@@ -600,6 +626,9 @@ def split(
                 "doesn't equal the number of splits {n}."
             )
 
+        blocks = self._plan.execute()
```

Comment:
Every function that triggers lazy execution (other than obvious ones like `.fully_executed()` or consuming functions like `.iter_batches()`) should note that it does so in its docstring. Do you want to wait until lazy mode is de-experimentalized before adding that to the appropriate docstrings?

Reply:
Yeah, I'm not even sure if we will expose lazy mode in the API, so let's leave it for now.